app/src/main/java/org/fossasia/pslab/communication/sensors/AD9833.java

```diff
@@ -0,0 +1,106 @@
+package org.fossasia.pslab.communication.sensors;
+
+import android.hardware.usb.UsbManager;
+import android.util.Log;
+import org.fossasia.pslab.communication.ScienceLab;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+
+/**
+ * Created by akarshan on 4/23/17.
+ * <p>
+ *     Instance of UsbManager is required to passes to the AD9833 constructor.
+ * </p>
+ */
+
+public class AD9833 {
+    //control bytes
+    private static final String TAG = "AD9833";
+    private int DDS_MAX_FREQ = 0xFFFFFFF-1;
+    private int DDS_B28 = 13;
+    private int DDS_HLB = 12;
+    private int DDS_FSELECT = 11;
+    private int DDS_PSELECT = 10;
+    private int DDS_RESET = 8;
+    private int DDS_SLEEP1 = 7;
+    private int DDS_SLEEP12 = 6;
+    private int DDS_OPBITEN = 5;
+    private int DDS_DIV2 = 3;
+    private int DDS_MODE = 1;
+
+    private int DDS_FSYNC = 9;
+
+    private int[] DDS_SINE = {0};
+    private int DDS_TRIANGLE = (1 << DDS_MODE);
+    private int DDS_SQUARE = (1 << DDS_OPBITEN);
+    private int DDS_RESERVED = (1 << DDS_OPBITEN) | (1 << DDS_MODE);
+    private int  clockScaler = 4;       // 8MHz
+    private int waveformMode;
+    private int activeChannel;
+    private int frequency;
+    private int cs;
+    private ScienceLab scienceLab;
+    
+    public AD9833(UsbManager usbManager) throws IOException{        //this approach needs to be checked.
+        scienceLab = new ScienceLab(usbManager);
```

Comment:
Don't create a new scienelab object, if needed I would pass sciencelab object from activity ( or wherever I will make object of this class ).
What i am thinking is I would pass sciencelab.spi to AD983's constructor and I suppose mapReferenceClock is only sciencelab method used, so would call it before making AD983's object.
Works?